autoPyTorch/evaluation/tae.py

```diff
@@ -190,6 +177,29 @@ def __init__(
 
         self.search_space_updates = search_space_updates
 
+        if isinstance(self.resampling_strategy, (HoldoutValTypes, CrossValTypes)):
+            eval_function = eval_train_function
+        elif isinstance(self.resampling_strategy, NoResamplingStrategyTypes):
+            eval_function = eval_test_function
+            self.output_y_hat_optimization = False
+
+        self.worst_possible_result = cost_for_crash
+
+        eval_function = functools.partial(
+            fit_predict_try_except_decorator,
+            ta=eval_function,
+            cost_for_crash=self.worst_possible_result,
+        )
+
+        super().__init__(
+            ta=ta if ta is not None else eval_function,
+            stats=stats,
+            run_obj=run_obj,
+            par_factor=par_factor,
+            cost_for_crash=self.worst_possible_result,
+            abort_on_first_run_crash=abort_on_first_run_crash,
+        )
+
```

Comment:
Can we get it back to the original place, because most variables input in `super()` are temporal variables and I would like to make sure that those will not be updated while we initialize other attributes.

Reply:
It sounds unsafe to depend on the order of the initialization of the parent class, but because `TestEvaluator` can take `output_y_hat_optimization` as a keyword argument, how about passing it as an argument?

Reply:
yes that is exactly happening. This `self.output_y_hat_optimization` is passed to the test evaluator.

Reply:
I have addressed your earlier comment though